### PR TITLE
GDPR-364: 🐛 Ensure deceased movement is included in the deceased deletion job result

### PR DESCRIPTION
**Pull request by hyderkhanmoj**

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerdeletionservice/service/DeceasedOffenderDeletionServiceTest.kt [hunk 1]
```diff
@@ -157,11 +157,11 @@ internal class DeceasedOffenderDeletionServiceTest {
       .thenReturn(setOf(offenderId2))
 
     whenever(
-      movementsService.getMovementsByOffenders(listOf(offenderNumber1), listOf("DEC"), true, true)
-    ).thenReturn(emptyList())
+      movementsService.getDeceasedMovementByOffenders(listOf(offenderNumber1))
+    ).thenReturn(null)
```
Comment (hyderkhanmoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerdeletionservice/service/DeceasedOffenderDeletionServiceTest.kt:161]: Done to be explicit here

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerdeletionservice/repository/model/Movement.kt [hunk 2]
```diff
@@ -45,4 +47,12 @@ data class Movement(
   val movementReason: String? = null,
 
   val commentText: String? = null
-)
+) {
+  fun formattedCopy() =
+    this.copy(
+      fromAgencyDescription = this.fromAgencyDescription?.formatLocation()?.trim() ?: "",
+      toAgencyDescription = this.toAgencyDescription?.formatLocation()?.trim() ?: "",
+      toCity = this.toCity?.trim() ?: "".capitalize(),
```
[the remark this answers is not in the document]
Comment (hyderkhanmoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerdeletionservice/repository/model/Movement.kt:55]: it was - will update now

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerdeletionservice/e2e/DataComplianceMessageTest.kt [hunk 3]
```diff
@@ -413,46 +413,46 @@ class DataComplianceMessageTest : IntegrationTestBase() {
...
+               "offenderId":-1023,
+               "offenderBookIds":[
+                  
                ]
             }
+         ]
+      },
+      {
+         "offenderIdDisplay":"Z0017ZZ",
+         "firstName":"MICHEAL",
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerdeletionservice/e2e/DataComplianceMessageTest.kt:438]: Micheal - my pet hate 🤣

File: src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerdeletionservice/repository/MovementsRepositoryTest.kt [hunk 4]
```diff
@@ -40,4 +40,17 @@ class MovementsRepositoryTest {
     val movements: List<Movement> = repository.getMovementsByOffenders(listOf("A6676RS"), listOf(), false, false)
     assertThat(movements).extracting<String>(Movement::fromAgency).containsOnly("BMI", "LEI")
   }
+
+  @Test
+  fun `can retrieve deceased movement by offenders`() {
```
Comment (mikehalmamoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerdeletionservice/repository/MovementsRepositoryTest.kt:45]: You could make this a `@ParameterizedTest` passing in the prison number and expected results which would remove the duplicated assertions.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerdeletionservice/repository/model/Movement.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (hyderkhanmoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerdeletionservice/repository/model/Movement.kt]: Updated, I will add an extra unit test in a subsequent commit

**Review by mikehalmamoj — APPROVED** [review]

**Pull request merged**